test/common/index.js

```diff
@@ -842,6 +842,36 @@ function requireNoPackageJSONAbove(dir = __dirname) {
   }
 }
 
+function spawnPromisified(...args) {
+  let stderr = '';
+  let stdout = '';
+
+  const child = spawnAsync(...args);
+  child.stderr.setEncoding('utf8');
+  child.stderr.on('data', (data) => { stderr += data; });
```

Comment:
consider using Readable stream `toArray`

Reply:
Oow, I'm not familiar with that; would it be `const stderr = child.stderr.toArray().join('');`? Is that more efficient? (It looks 6 of one, half a dozen of the other)

Reply:
it does the same thing with less code I guess,
something like this seems more readable and simple to me: 
```js
  const child = spawn(...args);
  child.stderr.setEncoding('utf8');
  child.stdout.setEncoding('utf8');

  const [stdout, stderr, [code, signal], error] = await Promise.all([
		child.stdout.toArray(),
    child.stderr.toArray(),
		once(child, 'exit'),
		Promise.race(once(child, 'error'), once(child, 'exit').then(()=>{}))
  ]);
```

Reply:
I would think that `toArray().join('')` uses more memory, as the entire output gets assembled as an array first and then reassembled as a string, whereas `+= data` is more efficient. For Node’s tests the outputs are probably always so small that we wouldn’t notice, but I would think that any common code like a helper should be somewhat optimized for performance.

Reply:
I'm not sure this is better, especially because I don't understand it.

I'm trying to troubleshoot the error that's producing, and the problem I'm on now is that `stderr` and `stdout` are arrays without `join()` methods.

Reply:
just a suggestion, I understand why this might not be as performant, but IMHO it is a bit more readable since it includes less indentation and less callbacks

Reply:
Thanks! I appreciate the suggestion and to learn new tools, but I think the original is much more straightforward. This is a bit more consolidated, but I think not in a good way. Since this is less performant and less straightforward, I'd rather not spend my night troubleshooting the errors it's producing.

Reply:
And also, `.toArray` is not available on Node.js 16.x IIRC, so that would make backporting harder (although one could say the same thing about `node:test`).